v2/tools/generator/internal/codegen/pipeline/handle_float_values.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// HandleFloatValuesStageID is the unique identifier of this stage
+const (
+	HandleFloatValuesStageID = "handleFloatValues"
+)
+
+func HandleFloatValues() *Stage {
+	stage := NewStage(
+		HandleFloatValuesStageID,
+		"Transform Float type values to Integer types",
+		func(ctx context.Context, state *State) (*State, error) {
+
+			definitions := state.Definitions()
+			specDefinitions := astmodel.FindSpecDefinitions(definitions)
+
+			result, errs := getFloatTransformations(definitions, specDefinitions)
+			if err := kerrors.NewAggregate(errs); err != nil {
+				return nil, err
+			}
+
+			remaining := definitions.Except(result)
+			result.AddTypes(remaining)
+
+			return state.WithDefinitions(result), nil
+		})
+
+	stage.RequiresPrerequisiteStages(RemoveStatusPropertyValidationsStageID)
+
+	return stage
+}
+
+func getFloatTransformations(definitions astmodel.TypeDefinitionSet, validatedDefinitions astmodel.TypeDefinitionSet) (astmodel.TypeDefinitionSet, []error) {
+	walker := astmodel.NewTypeWalker(
+		definitions,
+		astmodel.TypeVisitorBuilder{
+			VisitObjectType: transformFloatToInt,
```

Comment:
Instead of doing this on `VisitObjectType` where you have to iterate the properties, etc, I think you can just write this:

```
VisitValidatedType: transformFloatToInt,
```

and then:

```
func transformFloatToInt(this *astmodel.TypeVisitor, validated *astmodel.ValidateType, ctx interface{}) (astmodel.Type, error) {
	if result, ok := astmodel.AsPrimitiveType(validated.ElementType()); ok && result == astmodel.FloatType {
		validated = validated.WithElement(astmodel.IntType)
	}
	astmodel.IdentityVisitOfValidatedType(this, validated, ctx)
}
```

I also don't think you need the `MakeTypeOptional` bit?

Actually you probably ought to set a context instead of changing the type right in the `VisitValidatedType`, so do:
```
func visitValidatedType(this *astmodel.TypeVisitor, validated *astmodel.ValidateType, ctx interface{}) (astmodel.Type, error) {
	astmodel.IdentityVisitOfValidatedType(this, validated, true) // Pass ctx so that transformFloatToInt can use it
}

func transformFloatToInt(this *astmodel.TypeVisitor, prim *astmodel.PrimitiveType, ctx interface{}) (astmodel.Type, error) {
	validated, ok := ctx.(bool)
	if prim == astmodel.FloatType && validated {
		return astmodel.IntType
	}
	astmodel.IdentityVisitOfPrimitiveType(this, validated, ctx)
}
```